File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number[]}
     */
    productExceptSelf(nums) {

        var runningProduct = 1;
        var prefixProducts = [];

        for (var i = 0; i < nums.length; i++) {
          prefixProducts[i] = runningProduct;
          runningProduct *= nums[i];
        }

        var suffixProduct = 1;
        var output = prefixProducts;
        for(var i = nums.length -1; i >= 0; i--){
            output[i] *= suffixProduct;
            suffixProduct *= nums[i];
        }

        return output;
    }
}

File: Data Structures & Algorithms/concatenation-of-array/submission-1.java
class Solution {
    public int[] getConcatenation(int[] nums) {
        int newLength = nums.length * 2;
        int [] newArray = new int[newLength];

        for(int i = 0; i < newLength; i++){
            if(i >= nums.length){
                newArray[i] = nums[i - nums.length];
            }else{
                newArray[i] = nums[i];
            }

        }
        return newArray;
    }
}